### PR TITLE
Bluetooth: Mesh: support unknown range in ctl

### PR DESCRIPTION
**Pull request by alxelax**
Commit adds support of unknown temperature range
for CTL model.

### CODE REVIEW

File: subsys/bluetooth/mesh/light_temp_srv.c [hunk 1]
```diff
@@ -466,22 +475,25 @@ void bt_mesh_light_temp_srv_set(struct bt_mesh_light_temp_srv *srv,
...
-				 struct bt_mesh_light_temp_range *range)
+enum bt_mesh_model_status bt_mesh_light_temp_srv_range_set(struct bt_mesh_light_temp_srv *srv,
+							   struct bt_mesh_msg_ctx *ctx,
+							   struct bt_mesh_light_temp_range *range)
 {
 	const struct bt_mesh_light_temp_range old = srv->range;
 
-	if ((range->min < BT_MESH_LIGHT_TEMP_MIN) ||
-	    (range->min > range->max)) {
+	if (range->min < BT_MESH_LIGHT_TEMP_MIN && range->min != BT_MESH_LIGHT_TEMP_UNKNOWN) {
```
Comment (ludvigsj) [inline subsys/bluetooth/mesh/light_temp_srv.c:484]: I can't seem to get this to make sense ... wouldn't `range->min < TEMP_MIN` mean that `range->min != TEMP_UNKNOWN` is always true?

File: subsys/bluetooth/mesh/light_ctl_internal.h [hunk 2]
```diff
@@ -16,24 +16,34 @@
...
-	return srv->range.min +
-	       ROUNDED_DIV((int32_t)(lvl + LIGHT_TEMP_LVL_OFFSET) *
-				(srv->range.max - srv->range.min),
-				(int32_t)UINT16_MAX);
+	return min == BT_MESH_LIGHT_TEMP_UNKNOWN ? BT_MESH_LIGHT_TEMP_MIN : min;
 }
 
-static inline int16_t temp_to_lvl(struct bt_mesh_light_temp_srv *srv,
-				  uint16_t raw_temp)
+static inline uint16_t get_range_max(uin16_t max)
```
Comment (akredalen) [inline subsys/bluetooth/mesh/light_ctl_internal.h:24]:
```suggestion
static inline uint16_t get_range_max(uint16_t max)
```

File: subsys/bluetooth/mesh/light_ctl_internal.h [hunk 3]
```diff
@@ -16,24 +16,34 @@
 
 #define LIGHT_TEMP_LVL_OFFSET (-INT16_MIN)
 
-static inline uint16_t lvl_to_temp(struct bt_mesh_light_temp_srv *srv,
-				   int16_t lvl)
+static inline uint16_t get_range_min(uin16_t min)
```
Comment (akredalen) [inline subsys/bluetooth/mesh/light_ctl_internal.h:19]:
```suggestion
static inline uint16_t get_range_min(uint16_t min)
```

**Review by ludvigsj — APPROVED** [review]

**Review by akredalen — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/light_ctl_internal.h (hunk 2 above)
Comment (alxelax) [reply subsys/bluetooth/mesh/light_ctl_internal.h]: Ooops. :(  fixed

**Review by akredalen — APPROVED** [review]

**Pull request merged**